lib/inc/cpp-pcp-client/connector/connector_base.hpp

```diff
@@ -0,0 +1,197 @@
+#pragma once
+
+#include <cpp-pcp-client/connector/connection.hpp>
+#include <cpp-pcp-client/connector/client_metadata.hpp>
+#include <cpp-pcp-client/connector/errors.hpp>
+#include <cpp-pcp-client/connector/timings.hpp>
+
+#include <cpp-pcp-client/protocol/parsed_chunks.hpp>
+
+#include <cpp-pcp-client/validator/validator.hpp>
+#include <cpp-pcp-client/validator/schema.hpp>
+
+#include <cpp-pcp-client/util/thread.hpp>
+
+#include <cpp-pcp-client/export.h>
+
+namespace PCPClient {
+
+//
+// ConnectorBase
+//
+// A base class for shared implemention between v1 and v2 connectors.
+//
+
+class LIBCPP_PCP_CLIENT_EXPORT ConnectorBase {
+  public:
+    using MessageCallback = std::function<void(const ParsedChunks&)>;
+
+    ConnectorBase() = delete;
+
+    ConnectorBase(std::vector<std::string> broker_ws_uris,
+        std::string client_type,
+        std::string ca_crt_path,
+        std::string client_crt_path,
+        std::string client_key_path,
+        long ws_connection_timeout_ms,
+        uint32_t association_timeout_s,
+        uint32_t association_request_ttl_s,
```

Comment:
You would have to modify `ClientMetadata` to move the two association time periods arguments and related class members out of the base class. I'm not sure if it's worth.
By the way, those two values are only used by the `Connector`.